src/Microsoft.Health.Fhir.Api/Registration/FhirServerApplicationBuilderExtensions.cs

```diff
@@ -23,10 +27,56 @@ public static IApplicationBuilder UseFhirServer(this IApplicationBuilder app)
 
             app.UseHealthChecksExtension(new PathString(KnownRoutes.HealthCheck));
 
+            var config = app.ApplicationServices.GetService(typeof(IOptions<FhirServerConfiguration>)) as IOptions<FhirServerConfiguration>;
+            EnsureArg.IsNotNull(config, nameof(config));
+
+            var pathBase = new PathString(config.Value.PathBase.TrimEnd('/'));
+            if (pathBase.HasValue)
+            {
+                app.UseMiddleware<PathBaseMiddleware>(pathBase);
```

Comment:
.NET also has https://docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.builder.usepathbaseextensions.usepathbase?view=aspnetcore-3.1, does this built-in function that looks like it does something very similar

Reply:
Correct. But it does something slightly different.
Here is an example with PathBase set to `/r4`.
| Request | PR | .NET | Comment | 
|-|-|-|-|
| `/Patient` | PathBase: `/r4`<br/>Path: `/Patient` | PathBase: `null`<br/>Path: `/Patient` | PR version is what is needed for rewriting proxy.<br/>.NET version makes no changes to this request |
| `/r4/Patient` | PathBase: `/r4`<br/>Path: `/r4/Patient`<br/>Will result in **404** | PathBase: `/r4`<br/>Path: `/Patient` | PR version assumes original request was `/r4/r4/Patient`.<br/>.NET version performs a rewrite and routes to `/Patient` route |

I can add logic to handle the second scenario the way .NET version does, but .NET version is not suitable to handle the first scenario.

Here is a link to a relevant source: https://github.com/dotnet/aspnetcore/blob/master/src/Http/Http.Abstractions/src/Extensions/UsePathBaseMiddleware.cs#L54-L74

Reply:
@brendankowitz should I make server respond on both `/` and `/{{pathBase}}`?